### ISSUE TITLE
M2.3.2 : Nullable getters in Service Contracts will throw a reflection error when used in the web API 

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#25656]
### Preconditions

1. Magento 2.3.2 & 2.3-develop

### Steps to reproduce

Create an interface with a nullable getter. In our scenario it was a method that could return an array of instances of another object, or `null`:

```php
/**
 * @return null|\Vendor\Module\Api\Data\CompanyAddressInterface[]
 */
public function getAddresses(): ?array;
```

### Expected result

When properly configured and used as a web API, the object should be serialized using Reflection and it's nested objects as well.

### Actual result

The following error is thrown:

![image](https://user-images.githubusercontent.com/51680850/69247499-2968b180-0bb3-11ea-9f73-7be2c423c988.png)

### Possible cause

My guess is that when determinating the type of return class, something goes wrong when a return type can be nullable. See also the workaround:

### Workaround

The current woraround that will not trigger this error and make the web API work as expected is to flip the arguments, so instead of:

```php
/**
 * @return null|\Vendor\Module\Api\Data\CompanyAddressInterface[]
 */
public function getAddresses(): ?array;
```

I use:

```php
/**
 * @return \Vendor\Module\Api\Data\CompanyAddressInterface[]|null
 */
public function getAddresses(): ?array;
```

Now everything works as expected, but it seems to me that this should be considered a bug.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29037` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by theCapypara** [comment]
A related issue, which must also be fixed:
When generating the swagger schema, Magento fails, if null is the first type in the ``@return`` annotation:
```
<response>
<message>
The "" data type isn't declared. Verify the type and try again.
</message>
<trace>
#0 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(471): Magento\Framework\Reflection\TypeProcessor->getTypeData('') #1 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(449): Magento\Webapi\Model\Rest\Swagger\Generator->generateDefinition('') #2 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(476): Magento\Webapi\Model\Rest\Swagger\Generator->getObjectSchema(NULL, '') #3 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(449): Magento\Webapi\Model\Rest\Swagger\Generator->generateDefinition('TudockTestDataT...') #4 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(910): Magento\Webapi\Model\Rest\Swagger\Generator->getObjectSchema('TudockTestDataT...', '') #5 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(271): Magento\Webapi\Model\Rest\Swagger\Generator->generateMethodSuccessResponse(Array, Array) #6 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(252): Magento\Webapi\Model\Rest\Swagger\Generator->generateMethodResponses(Array) #7 /src/app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php(170): Magento\Webapi\Model\Rest\Swagger\Generator->generatePathInfo('get', Array, 'tudockTestV1') #8 /src/app/code/Magento/Webapi/Model/AbstractSchemaGenerator.php(109): Magento\Webapi\Model\Rest\Swagger\Generator->generateSchema(Array, 'https', 'magento-github....', '/rest/all/schem...') #9 /src/app/code/Magento/Webapi/Controller/Rest/SchemaRequestProcessor.php(59): Magento\Webapi\Model\AbstractSchemaGenerator->generate(Array, 'https', 'magento-github....', '/rest/all/schem...') #10 /src/app/code/Magento/Webapi/Controller/Rest.php(188): Magento\Webapi\Controller\Rest\SchemaRequestProcessor->process(Object(Magento\Framework\Webapi\Rest\Request\Proxy)) #11 /src/lib/internal/Magento/Framework/Interception/Interceptor.php(58): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http)) #12 /src/lib/internal/Magento/Framework/Interception/Interceptor.php(138): Magento\Webapi\Controller\Rest\Interceptor->___callParent('dispatch', Array) #13 /src/lib/internal/Magento/Framework/Interception/Interceptor.php(153): Magento\Webapi\Controller\Rest\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http)) #14 /src/generated/code/Magento/Webapi/Controller/Rest/Interceptor.php(26): Magento\Webapi\Controller\Rest\Interceptor->___callPlugins('dispatch', Array, Array) #15 /src/lib/internal/Magento/Framework/App/Http.php(116): Magento\Webapi\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http)) #16 /src/generated/code/Magento/Framework/App/Http/Interceptor.php(24): Magento\Framework\App\Http->launch() #17 /src/lib/internal/Magento/Framework/App/Bootstrap.php(260): Magento\Framework\App\Http\Interceptor->launch() #18 /src/pub/index.php(40): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor)) #19 {main}
</trace>
</response>
```

**Comment by magento-engcom-team** [comment]
Hi @kanduvisla. Thank you for your report. 
The issue has been fixed in magento/magento2#25806 by @Parakoopa in 2.4-develop branch
Related commit(s):
 - [0cea191adfc16bb7b3fcfc5b0d331bf3420ecc5d](https://github.com/magento/magento2/commit/0cea191adfc16bb7b3fcfc5b0d331bf3420ecc5d)
 - [99a1a98228b4dc0e280ad9d3feec89f6d2ecb334](https://github.com/magento/magento2/commit/99a1a98228b4dc0e280ad9d3feec89f6d2ecb334)
 - [253f881cb1c4b8ca1a8d8a94d939f9b76f1cd6b2](https://github.com/magento/magento2/commit/253f881cb1c4b8ca1a8d8a94d939f9b76f1cd6b2)
 - [7e3390dd86faf45018a18e5f76d0ebe067019186](https://github.com/magento/magento2/commit/7e3390dd86faf45018a18e5f76d0ebe067019186)

The fix will be available with the upcoming 2.4.0 release.
